src/components/controls/Icon.js

```diff
@@ -241,6 +242,105 @@ export default class C extends Component<Props, State> {
             case 'namespace_light':
                 source = require('@src/assets/icons/namespace.png');
                 break;
+
+            case `transaction_${TransactionType.TRANSFER}`:
```

Comment:
This switch statement was too big and it's getting bigger imo. Should we consider replacing it with an associative array instead? Something like this:

```
let src = '';
const iconArray = [];
iconArray[`none`] = '';
iconArray[`transaction_${TransactionType.TRANSFER}`] = '@src/assets/icons/transaction/transfer.png';
iconArray[`transaction_${TransactionType.TRANSFER}_incoming`] ='@src/assets/icons/transaction/incoming.png';
iconArray[`transaction_${TransactionType.TRANSFER}_outgoing`] = '@src/assets/icons/transaction/outgoing.png'
...
// end of iconArray fill
src = iconArray[name] || '@src/assets/icons/ic-about.png';
if (src) {
    source = require(src);
}
// rest of the function
```

Reply:
Seems it is not possible to do this way.

> the image name in require has to be known statically.

This is from React Native docs (https://reactnative.dev/docs/images#static-image-resources)

```
// GOOD
<Image source={require('./my-icon.png')} />;

// BAD
var icon = this.props.active
  ? 'my-icon-active'
  : 'my-icon-inactive';
<Image source={require('./' + icon + '.png')} />;

// GOOD
var icon = this.props.active
  ? require('./my-icon-active.png')
  : require('./my-icon-inactive.png');
<Image source={icon} />;


Another option is to import them all together. But need to ensure it will not cause any performance issue:

const iconSrcMap = {
    [`transaction_${TransactionType.HASH_LOCK}`]: require('@src/assets/icons/transaction/lock.png'),
    [`transaction_${TransactionType.SECRET_PROOF}`]: require('@src/assets/icons/transaction/cross-chain.png'),
    [`transaction_${TransactionType.VRF_KEY_LINK}`]: require('@src/assets/icons/transaction/key.png'),
    [`transaction_${TransactionType.ACCOUNT_KEY_LINK}`]: require('@src/assets/icons/transaction/key.png'),
    [`transaction_${TransactionType.NODE_KEY_LINK}`]: require('@src/assets/icons/transaction/key.png'),
    [`transaction_${TransactionType.VOTING_KEY_LINK}`]: require('@src/assets/icons/transaction/key.png'),
    ...
}

return <Image source={iconSrcMap[name]} />
```

Reply:
I see, better to leave it as it is then 👍